src/app/layout/app-info/reminder-timeline/reminder-timeline.component.ts

```diff
@@ -0,0 +1,228 @@
+        lineWidth: 10,
+        name: {
+          title: 'Recreation',
+          icon: 'icon-keyboard'
+        }
+      }, {
+        data: [[1499436458000, 2], [1499447258000, 2]],
+        lineWidth: 10,
+        name: {
+          title: 'Visit',
+          icon: 'icon-stethoscope'
+        }
+      }, {
+        data: [[1499466393000, 5], [1499469993000, 5]],
+        lineWidth: 10,
+        name: {
+          title: 'Work',
+          icon: 'icon-briefcase'
+        }
+      }, {
+        data: [[1499473593000, 2], [1499484393000, 2]],
+        lineWidth: 10,
+        name: {
+          title: 'Shopping',
+          icon: 'icon-shopping-bag'
+        }
+      }, {
+        data: [[1499477193000, 5], [1499491593000, 5]],
+        lineWidth: 10,
+        name: {
+          title: 'Confort',
+          icon: 'icon-flash'
+        }
+      }]
+    };
+  }
+
+  timelineSeries(count = 8) {
+    const offsets = shuffle(times(count, (index) => index));
+    const labels = [['Confort', 'icon-bed'],
+    ['Eco', 'icon-leaf'],
+    ['Night', 'icon-moon'],
+    ['Game', 'icon-gamepad'],
+    ['Recreation', 'icon-keyboard'],
+    ['Visit', 'icon-stethoscope'],
+    ['Work', 'icon-briefcase'],
+    ['Shopping', 'icon-shopping-bag']];
+    return times(count, (index) => {
+      const offset = offsets.pop();
+      const title = labels.pop();
+      const date = new Date().getTime();
+      const seeek = random(-72, 72);
+      const begin = date + (seeek * 10000);
+      return {
+        data: [
+          [begin, offset % 2 === 0 ? offset : offset + 2],
+          [begin + (random(1, 3) * 3600 * 1000), offset % 2 === 0 ? offset : offset + 2]
+        ],
+        lineWidth: 10,
+        name: {
+          title: title[0],
+          icon: title[1]
+        }
+      };
+    });
+  }
+
+  titleGenerator() {
+    const _this = this;
+    for (const k in _this.chart.series) {
+      if (typeof _this.chart.series[k] === 'object') {
+        const icon = '<i class="icon ' + _this.chart.series[k].name.icon + '"></i>',
+            title = '<span>' + _this.chart.series[k].name.title + '</span>',
+            titleWrapp = document.createElement('span');
+        titleWrapp.innerHTML = icon + title;
+        titleWrapp.id = 'highcharts-series-title' + k;
+        titleWrapp.style.position = 'absolute';
+        titleWrapp.style.color = _this.chart.series[k].color;
+        titleWrapp.style.fontSize = '14';
+        titleWrapp.style.fontWeight = 'bold';
+        titleWrapp.style.width = '100%';
+        titleWrapp.style.left = (_this.retPosition()[k].x + (_this.retPosition()[k].width / 3)) + 'px';
+        titleWrapp.style.top = (_this.retPosition()[k].y + 7) + 'px';
+        // tslint:disable-next-line:max-line-length
+        _this.retPosition()[k].x + (_this.retPosition()[k].width / 3) <= 0 ? titleWrapp.style.opacity = '0' : titleWrapp.style.opacity = '1';
+        document.getElementById('timeline').appendChild(titleWrapp);
+      }
+    }
+  }
```

Comment:
This function is really weird, please add some explanation for function, and refactor it a bit.